cli/src/semgrep/config_resolver.py

```diff
@@ -192,15 +198,20 @@ def _load_config_from_local_path(self) -> Dict[str, YamlTree]:
 
     def _make_config_request(self) -> str:
         app_session = get_state().app_session
-        r = app_session.get(
+        resp = app_session.get(
             self._config_path,
             headers={"Accept": "application/json", **self._extra_headers},
         )
-        if r.status_code == requests.codes.ok:
-            return r.content.decode("utf-8", errors="replace")
+        if resp.status_code == requests.codes.ok:
+            try:
+                rules_dict = resp.json()["rule_config"]
+                assert rules_dict is not None
+                return pyyaml.dump(rules_dict, Dumper=SafeDumper)  # type: ignore
```

Comment:
moving my slack message here:

> by the way I see you're doing this to write our rules: pyyaml.dump(rules_dict)
do you think we need to use YAML here? if we just json.dump instead then maybe we can skip all the trouble of adding a dependency

Reply:
JSON is also faster to dump and load, by the way. And YAML is compatible with everything JSON supports, so even if somewhere down do line we yaml.load the same contents, it'll still work. But I think everything downstream will load rules as JSON and fall back to YAML only if needed.

Reply:
It seems like all the schema validation stuff is done in Yaml and breaks with JSON. All the happy-path cases work fine with JSON, but the tests for bad_config fail because _validate() does not return any errors even if a required field like 'pattern' is missing. We could try to re-write this to be JSON-compatible but that feels like scope-creep to me 🤷

Reply:
Hmm that's weird. Looks like this function returns a string so it shouldn't matter if that string was generated by yaml or json library?

Reply:
Well it returns different strings - one looks like yaml (with - and \n all over the place) and the other looks like json (with { } and [ ])

Reply:
So seems like backend is returning rule_config as a json blob, then we parse that json as yaml and then dump it as a string. Should backend return it as a yaml string instead?

Reply:
It can't, because it is now returning all of the GET information at once. So it returns the rules yaml along with a bunch of other things that have always been returned as JSON. Does that make sense?

Reply:
for the record, a blob "with { } and [ ]"  is still valid YAML and there's no reason it should fail YAML validation 🤔

Reply:
@chmccreery that makes sense but no reason the value of a specific key in a json blob has to be json vs string. Can you link to where this is happening I can take a look